lib/Dialect/TritonGPU/Transforms/UpdateMmaForVolta.cpp

```diff
@@ -33,12 +34,13 @@ class CollectMmaToUpdateForVolta : public mlir::RewritePattern {
   mlir::LogicalResult
   matchAndRewrite(mlir::Operation *op,
                   mlir::PatternRewriter &rewriter) const override {
-
     auto dotOp = cast<triton::DotOp>(op);
```

Comment:
I don't understand why "op" is for sure to be a "triton::DotOp" here? Since in line 35, you are using mlir::Operation* as argument type. Did i miss something?

Reply:
[Here](https://github.com/openai/triton/blob/master/lib/Dialect/TritonGPU/Transforms/UpdateMmaForVolta.cpp#L30) it assigns DotOp's name to the RewritePattern's constructor, which means it should only trigger this pattern when visiting a DotOp instance.